medusa/search/core.py

```diff
@@ -485,13 +487,23 @@ def search_for_needed_episodes(force=False):
                     if first_result['date_added'] + cur_provider.fail_over_hours * 3600 > int(time.time()):
                         # The provider's fail over cooldown time hasn't expired yet. We're holding back the snatch.
                         log.debug(
-                            u'Holding back best result {name} for provider {provider}. The provider is waiting'
+                            u'DELAY: Holding back best result {name} for provider {provider}. The provider is waiting'
                             u' {fail_over_hours} hours, before accepting the release.', {'name': first_result['name'],
                                                                                          'provider': cur_provider.name,
                                                                                          'fail_over_hours': cur_provider.fail_over_hours
                                                                                          }
                         )
                         continue
+                    else:
+                        log.debug(
+                            u'DELAY: Provider {provider}, found a result in cache, but the delay has not yet expired. '
+                            u'Timestamp of first result: {first_result}. Time left: {left} hours',
+                            {'provider': cur_provider.name, 'first_result': first_result['date_added'],
+                             'left': int((int(time.time()) + (first_result['date_added'] - cur_provider.fail_over_hours * 3600)) / 3600)}
+                        )
+                else:
+                    log.debug(u'DELAY: Provider {provider}, searched cache but could not get any results for: {season}x{episode}',
```

Comment:
can you use the episode_num instead of this?

from medusa.helper.common import episode_num
episode_num(season=None, episode=None, numbering='standard'):

Reply:
also add show name